wasmtime-runtime/src/vmcontext.rs

```diff
@@ -422,14 +422,23 @@ impl VMSharedSignatureIndex {
     }
 }
 
+impl Default for VMSharedSignatureIndex {
+    fn default() -> Self {
+        VMSharedSignatureIndex::new(u32::MAX)
```

Comment:
Could you add a check in wasmtime-runtime where we allocate the shared signature table, to ensure that we don't ever create a table with 2**32 elements in it, so that `u32::MAX` is never a valid index?